src/datasources/entity-ds/datasource.ts

```diff
@@ -316,6 +319,11 @@ export class OpenNMSEntityDatasource {
   metricFindQuery(query, optionalOptions) {
     const options = optionalOptions || {};
 
+    const locations = query ? query.match(/locations\(.*\)/i) : null;
```

Comment:
FYI, `.*` is greedy by default, you may want to use "not close-paren" for the glob instead of `.` to avoid grabbing multiple functions, like so:
```suggestion
    const locations = query ? query.match(/locations\([^\)]*\)/i) : null;
```